tests/ui/specs/FanSpeedControl/FanSpeedControl-specs.js

```diff
@@ -0,0 +1,61 @@
+const Page = require('./FanSpeedControlPage');
+
+describe('FanSpeedControl', function () {
+	beforeEach(function () {
+		Page.open();
+	});
+
+	describe('default', function () {
+		const fanSpeedControl = Page.components.fanSpeedControlDefault;
+
+		it('should have the first arc selected by default', function () {
+			expect(fanSpeedControl.coloredPath(1).getCSSProperty('stroke').value).to.equal('rgb(68,68,68)');
+		});
+
+		it('should select all arcs up to the clicked arc', function () {
+			fanSpeedControl.clickablePath(3).click();
+			expect(fanSpeedControl.coloredPath(1).getCSSProperty('stroke').value).to.equal('rgb(68,68,68)');
+			expect(fanSpeedControl.coloredPath(2).getCSSProperty('stroke').value).to.equal('rgb(68,68,68)');
+			expect(fanSpeedControl.coloredPath(3).getCSSProperty('stroke').value).to.equal('rgb(68,68,68)');
+			// next one is unselected
+			expect(fanSpeedControl.coloredPath(4).getCSSProperty('stroke').value).to.equal('rgb(238,238,238)');
+		});
+
+		it('should display `fan` icon', function () {
```

Comment:
can you also have a FanSpeed component in the view with a custom icon and do 1 test to check the icon?

Reply:
sure